packages/client/src/sync/fetcher/fetcher.ts

```diff
@@ -226,17 +232,30 @@ export abstract class Fetcher<JobTask, JobResult, StorageItem> extends Readable
    * @param result job result
    */
   private success(job: Job<JobTask, JobResult, StorageItem>, result?: JobResult) {
-    if (job.state !== 'active') return
     let jobStr = this.jobStr(job, true)
+    if (job.state !== 'active') return
+
     let reenqueue = false
     let resultSet = ''
     if (result === undefined) {
       resultSet = 'undefined'
       reenqueue = true
     }
-    if (result !== undefined && (result as any).length === 0) {
-      resultSet = 'empty'
-      reenqueue = true
+    if (result !== undefined) {
+      if ('length' in (result as any)) {
```

Comment:
Any objection to re-writing this to something like `if (result !== null and result.length !== undefined)`?  I know its pedantic but I really don't like `as any` anywhere if we can help it.